### ISSUE TITLE
orb init should cope with things already existing

### ISSUE THREAD

**Issue by Peter-Darton-i2** [CircleCI-Public/circleci-cli#803]
TL;DR: The `orb init` tool needs to be more tolerant when finding that the things it needs to exist do already exist.

## Meta:

**CircleCI CLI Version:**
0.1.22322+5ff92b4 (release)

**Operating System:**
Ubuntu running under WSL2 on Win10

**CircleCI CLI Diagnostic:**
```
---
CircleCI CLI Diagnostics
---
Debugger mode: false
Config found: /home/pjd/.circleci/cli.yml
API host: https://circleci.com
API endpoint: graphql-unstable
OK, got a token.
Trying an introspection query on API...
Ok.
Hello, Peter Darton.
```

---

## Current behavior:
1. (Get a org admin to create a) new empty repo on github
1. `git clone git@github.com:myorg/circleci-orb-myorbname.git`
1. `cd circleci-orb-myorbname`
1. `circleci orb init . --private`
```
Yes
GitHub
myorg
myorg
myorbname
(no categories)
Yes
Yes
main
git@github.com:myorg/circleci-orb-myorbname.git
```
```
Thank you! Setting up your orb...
Error: repository already exists
```

## Expected behavior:
I expected this to be a valid way of invoking the tool.
1. If the folder is a git repo, even an empty one, the tool shouldn't need to ask for the git repo URL as that's already recorded in the .git folder; the tool should check for this and handle it.
1. It shouldn't report `Error:` if the folder has already been configured to point to a github repository - the tool should check for this and handle it.

The error message needs some clarification too.
1. It's not obvious that the error is a fatal error that caused the process to be abandoned; `git status` reported that the tool had made lots of changes to the local filesystem, implying that it'd actually done its job.
It was only from watching [the video](https://academy.circleci.com/orbs-course/954523/scorm/2sf5smsrh628p) that I realised that the tool should've continued on from that point.
1. The error message should specify that it's the _local_ repository that's being complained about.
I originally thought that it was upset that the repository had been created in github.
I know that, technically, the local .git folder is also a repository in its own right, but error messages need to be clear which repository they're referring to.

## When did this begin / Was this previously working?:
It's never worked for me.

## Additional Information:
There's many ways to create a repository and to get it linked to a local folder; different users will be used to different workflows that all result in the same end result ... but this tool _should not_ assume that a repo doesn't exist and _should_ cope when the folder it's working in is already pointing at a repository.
 - The tool _should_ pick up the existing remote github repository details from the .git folder and not need to ask.
 - The tool _should_ cope if there's existing files in the existing repository, possibly warning about it overwriting files.

Even if I use the tool _exactly_ as suggested in the video (i.e. letting the tool set up the local git repo), that didn't work either - instead of complaining `Error: repository already exists` it complained `Error: author field is required` and bombed out ... even though CDing to the folder and doing `git config` shows that user.name and user.email are both set.
i.e. it looks like #495 has come back.

...and when I tried re-running and when it asked "Would you like to set up your git project?" I answered "No", it then bombed out with the error "Error: Unable to create orb: Cannot create an Orb named 'myorbname': an Orb with that name already exists."
It's correct that it's been registered - my github admin has registered it - but the tool's inability to cope with that then means that, if anyone has ever run `circleci orb create ..." then that then prevents anyone from ever setting up development for the orb using the `orb init` tool.
i.e. it's a nasty (and permanent) either/or situation and there's no warning from `circleci orb create` that this is the case.

Note: If we ask it to "set up a publishing context for your orb" then it does successfully detect if it's done that already and it copes with that; that's how it should work everywhere.
i.e. what's needed is to extend this "create it only if it doesn't already exist" logic to all the other resource-creation code.

**Comment by pete-woods** [comment]
## Triage: partly fixed in the v1 CLI — narrowing this to the git half

The CLI has been rewritten for v1 (`main`; `v0` is the legacy CLI). Two of the three
things this report asked for are done:

- **Orb already registered** — `orb init` now looks the orb up first and reuses it,
  carrying on with the rest of the setup instead of failing. That was #804, closed.
- **Publishing context already exists** — now detected by looking the context up rather
  than by string-matching the API's error message, which is what made v0's "continuing on"
  path unreachable. That was #1192, closed.
- **Namespace** — created only if it does not already exist.

**Your actual scenario still fails**, so this stays open:

```console
$ git clone git@github.com:myorg/circleci-orb-myorbname.git
$ cd circleci-orb-myorbname
$ circleci orb init . --private
...
error: Could not initialize git repository: . is already a git repository; delete its .git directory and retry
```

Better message than `Error: repository already exists`, same dead end. Both of your points
hold:

1. If the directory is already a git repo, `orb init` should not ask for the remote URL —
   it is in `.git/config`. v1 already has a `gitremote` package that resolves exactly this,
   so the plumbing exists.
2. An existing repo should not be an error. Cloning an empty repo and initialising the orb
   into it is a reasonable — arguably the *obvious* — way to use this, and it is what an
   org where only admins create repositories forces you to do.

The fix is to make the git step follow the same check-first pattern as the namespace, orb
and context steps: if `.git` exists, adopt it — read `origin`, use the current branch as
the default, skip `git init` and `CreateRemote`, and go straight to staging the template
files and committing.
